test/gtest/common/test_perf.cc

```diff
@@ -199,29 +196,46 @@ test_perf::test_result test_perf::run_multi_threaded(const test_spec &test, unsi
     params.uct.am_hdr_size = 8;
     params.alignment       = ucs_get_page_size();
     params.max_outstanding = test.max_outstanding;
+
     if (ucs::test_time_multiplier() == 1) {
         params.warmup_iter = ucs_max(1, test.iters / 100);
         params.max_iter    = test.iters;
     } else {
         params.warmup_iter = 0;
         params.max_iter    = ucs_min(20u, test.iters /
-                                                  ucs::test_time_multiplier() /
-                                                  ucs::test_time_multiplier());
+                                          ucs::test_time_multiplier() /
+                                          ucs::test_time_multiplier());
     }
+
     params.max_time        = 0.0;
     params.report_interval = 1.0;
     params.rte_group       = NULL;
     params.rte             = &rte::test_rte;
     params.report_arg      = NULL;
+
     ucs_strncpy_zero(params.uct.dev_name, dev_name.c_str(), sizeof(params.uct.dev_name));
     ucs_strncpy_zero(params.uct.tl_name , tl_name.c_str(), sizeof(params.uct.tl_name));
+
     params.uct.data_layout = (uct_perf_data_layout_t)test.data_layout;
     params.uct.fc_window   = UCT_PERF_TEST_MAX_FC_WINDOW;
     params.msg_size_cnt    = test.msglencnt;
     params.msg_size_list   = (size_t *)test.msglen;
     params.iov_stride      = test.msg_stride;
     params.ucp.send_datatype = (ucp_perf_datatype_t)test.data_layout;
     params.ucp.recv_datatype = (ucp_perf_datatype_t)test.data_layout;
+}
+
+
+test_perf::test_result test_perf::run_multi_threaded(const test_spec &test, unsigned flags,
+                                                     const std::string &tl_name,
+                                                     const std::string &dev_name,
+                                                     const std::vector<int> &cpus)
+{
+    rte_comm c0to1, c1to0;
+    ucx_perf_params_t params;
+
+    memset(&params, 0, sizeof(params));
```

Comment:
why moved it here (instead of test_params_init)?

Reply:
didn't move, just leave it where it was - run_multi_threaded (and now single as well), it's a common approach - init func just for init only (it could not all of the fields) - so if user wants memset - it's on him...

Reply:
i think it will be a little less code duplication to keep memset in test_params_init(), so that test_params_init() will make sure the entire struct is initialized after it's called

Reply:
if we r talking about init, so ptr->some_field = 0; - this initialization and memset is not, because you set to zero fields that should get another value as well.. what if tomorrow we'll add some field that should be initialed out and before init func ?

Reply:
set the fields to 0 inside init function itself